src/v/cloud_storage/topic_manifest.cc

```diff
@@ -441,27 +436,6 @@ topic_manifest::update(manifest_format format, ss::input_stream<char> is) {
 
 ss::future<serialized_data_stream> topic_manifest::serialize() const {
     vassert(_topic_config.has_value(), "_topic_config is not initialized");
-
-    if (_manifest_version == first_version) {
```

Comment:
does this have backwards compatibility implications or implications on reading old data? the commit message mentions serde envelope version, but that seems like it is some what orthogonal to the span of manifest versions that have existed over time.

Reply:
We chatted in person, bringing it back here:

Not really. The in-memory state of the manifest is ultimately the topic configuration. When we serialize the JSON in, callers don't have any reason to treat it differently than if we serialized with serde